testing/integration/agent_long_test.go

```diff
@@ -0,0 +1,282 @@
+
+	runner.T().Logf("Installing fleet package....")
+	_, err = runner.info.KibanaClient.InstallFleetPackage(ctx, apachePackage)
+	require.NoError(runner.T(), err, "error creating fleet package")
+
+}
+
+func (runner *ExtendedRunner) TestHandleLeak() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	defer cancel()
+
+	testRuntime := os.Getenv("LONG_TEST_RUNTIME")
+	if testRuntime == "" {
+		testRuntime = "5m"
+	}
+
+	// because we need to separately fetch the PIDs, wait until everything is healthy before we look for running beats
+	require.Eventually(runner.T(), func() bool {
+		allHealthy := true
+		status, err := runner.agentFixture.ExecStatus(context.Background())
+		require.NoError(runner.T(), err)
+		for _, comp := range status.Components {
+			if comp.State != int(cproto.State_HEALTHY) {
+				allHealthy = false
+			}
+		}
+		return allHealthy
+	}, time.Minute*3, time.Second*20)
+
+	procs, err := sysinfo.Processes()
+	require.NoError(runner.T(), err)
+
+	handles := []processWatcher{}
+
+	// track running beats
+	// the `last 30s` metrics tend to report gauges, which we can't use for calculating a derivative.
+	// so separately fetch the PIDs
+	for _, proc := range procs {
+		info, err := proc.Info()
+		require.NoError(runner.T(), err)
+		if strings.Contains(info.Name, "beat") || strings.Contains(info.Name, "elastic-agent") {
+			handle, err := sysinfo.Process(proc.PID())
+			require.NoError(runner.T(), err)
+			handlesReg := new(regression.Regression)
+			handlesReg.SetObserved(fmt.Sprintf("%s handle usage", info.Name))
+			handlesReg.SetVar(0, "time")
+
+			runner.T().Logf("created handle watcher for %s (%d)", info.Name, proc.PID())
+			handles = append(handles, processWatcher{handle: handle, pid: proc.PID(), name: info.Name, regHandles: handlesReg})
+		}
+	}
+
+	testDuration, err := time.ParseDuration(testRuntime)
+	require.NoError(runner.T(), err)
+
+	timer := time.NewTimer(testDuration)
+
+	// time to perform a health check
+	ticker := time.Tick(time.Second * 10)
+
+	done := false
+	start := time.Now()
+	for !done {
+		select {
+		case <-timer.C:
+			done = true
+		case <-ticker:
+			err := runner.agentFixture.IsHealthy(ctx)
+			require.NoError(runner.T(), err)
+			// for each running process, collect memory and handles
+			for _, handle := range handles {
+
+				ohc, ok := handle.handle.(types.OpenHandleCounter)
+				if ok {
+					handleCount, err := ohc.OpenHandleCount()
+					require.NoError(runner.T(), err)
+					handle.regHandles.Train(regression.DataPoint(float64(handleCount), []float64{time.Since(start).Seconds()}))
+				}
+
+			}
+		}
+	}
+
+	// we're measuring the handle usage as y=mx+b
+	// if the slope is increasing above a certain rate, fail the test
+	handleSlopeFailure := float64(1)
+
+	for _, handle := range handles {
+		err = handle.regHandles.Run()
+		require.NoError(runner.T(), err)
+
+		runner.T().Logf("=============================== %s (%d)", handle.name, handle.pid)
```

Comment:
Just printing out the handle name isn't enough, we want the component name so we can tell if it a monitoring beat, the system metrics metricbeat, etc.

If you get the PIDs out of the elastic agent status command or state RPC as suggested above you can solve this.